x/evm/types/interfaces.go

```diff
@@ -33,3 +35,12 @@ type StakingKeeper interface {
 	GetHistoricalInfo(ctx sdk.Context, height int64) (stakingtypes.HistoricalInfo, bool)
 	GetValidatorByConsAddr(ctx sdk.Context, consAddr sdk.ConsAddress) (validator stakingtypes.Validator, found bool)
 }
+
+// Event Hooks
+// These can be utilized to customize evm transaction processing.
+
+// EvmHooks event hooks for evm tx processing
+type EvmHooks interface {
+	// Must be called after tx is processed, if failed, the whole evm transaction is reverted.
```

Comment:
Add the warning that we discussed in the `Negative` consequences section of the ADR

Reply:
Actually, that negative consequence is about the use case of utilizing the hook to process logs, not about the hook interface in general.

Reply:
I added the use case context to consequences in the adr.